src/plugins/controls/public/control_group/control_group_renderer.tsx

```diff
@@ -8,41 +8,87 @@
 
 import uuid from 'uuid';
 import useLifecycles from 'react-use/lib/useLifecycles';
-import React, { useEffect, useMemo, useRef, useState } from 'react';
+import React, { useMemo, useRef, useState } from 'react';
 
 import { IEmbeddable } from '@kbn/embeddable-plugin/public';
+import { useReduxContainerContext } from '@kbn/presentation-util-plugin/public';
 
 import { pluginServices } from '../services';
-import { getDefaultControlGroupInput } from '../../common';
-import { ControlGroupInput, ControlGroupOutput, CONTROL_GROUP_TYPE } from './types';
+import { ControlPanelState, getDefaultControlGroupInput } from '../../common';
+import {
+  ControlGroupInput,
+  ControlGroupOutput,
+  ControlGroupReduxState,
+  CONTROL_GROUP_TYPE,
+} from './types';
 import { ControlGroupContainer } from './embeddable/control_group_container';
+import { DataControlInput } from '../types';
+import { getCompatibleControlType, getNextPanelOrder } from './embeddable/control_group_helpers';
+import { controlGroupReducers } from './state/control_group_reducers';
+
+const ControlGroupInputBuilder = {
```

Comment:
The next step of this process will be to use an input builder like this one in the inline editor to serve the Control Creation use case. This is because the control editor in that case simply returns a partial input object for the Control type you're creating, so it would be cool to eventually use the same APIs for partial input creation inside or outside of the Controls plugin.